ui/components/component-library/button-icon/button-icon.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderRadius,
+  DISPLAY,
+  IconColor,
+  JustifyContent,
+} from '../../../helpers/constants/design-system';
+
+import Box from '../../ui/box';
+import { Icon } from '../icon';
+
+import { ButtonIconSize, ButtonIconProps } from './button-icon.types';
+
+export const ButtonIcon = React.forwardRef(
+  (
+    {
+      ariaLabel,
+      as = 'button',
+      className = '',
+      color = IconColor.iconDefault,
+      href,
+      size = ButtonIconSize.Lg,
+      iconName,
+      disabled,
+      iconProps,
+      ...props
+    }: ButtonIconProps,
+    ref: React.Ref<HTMLElement>,
+  ) => {
+    const Tag = href ? 'a' : as;
+    return (
+      <Box
+        aria-label={ariaLabel}
+        as={Tag}
+        className={classnames(
+          'mm-button-icon',
+          `mm-button-icon--size-${String(size)}`,
+          {
+            'mm-button-icon--disabled': Boolean(disabled),
+          },
+          className,
+        )}
+        color={color}
+        {...(disabled ? { disabled } : {})}
```

Comment:
No that we are limiting `as` to just `button` and `a` tags could we do something like
`const isDisabled = disabled && Tag === 'button';`
```suggestion
        ...(isDisabled ? { disabled: true } : {})} // only allow disabled attribute to be passed down to the Box when the as prop is equal to a button element
```